operator/src/main/java/org/keycloak/operator/Constants.java

```diff
@@ -33,10 +34,10 @@ public final class Constants {
     public static final String COMPONENT_LABEL = "app.kubernetes.io/component";
     public static final String KEYCLOAK_COMPONENT_LABEL = "keycloak.org/component";
 
-    public static final Map<String, String> DEFAULT_LABELS = Map.of(
+    public static final Map<String, String> DEFAULT_LABELS = new TreeMap<>(Map.of(
```

Comment:
Could you please elaborate a bit why to make this change? I suspect to make the iteration order predictable.

I'm not fully comfortable with making the Map modifiable, especially when we pass it around e.g. to Status. Maybe we could perhaps wrap it in `Collections.unmodifiableMap`?